hydro_deploy/core/src/kubernetes/mod.rs

```diff
@@ -0,0 +1,347 @@
+                #[cfg(unix)]
+                {
+                    self.id == id
+                }
+
+                #[cfg(not(unix))]
+                {
+                    let _ = id;
+                    false
+                }
+            }
+            // target_host.as_any().downcast_ref::<PodHost>()
+            ClientStrategy::InternalTcpPort(_target_host) => true, // TODO: if I'm on the same cluster, can just return true first
+            ClientStrategy::ForwardedTcpPort(_) => true,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct LaunchedPod {
+    pub internal_ip: String, // internal IP address for inter-pod communication in a node
+    pub pod_name: String,
+}
+
+#[async_trait]
+impl LaunchedHost for LaunchedPod {
+    fn server_config(&self, bind_type: &ServerStrategy) -> ServerBindConfig {
+        match bind_type {
+            ServerStrategy::UnixSocket => ServerBindConfig::UnixSocket,
+            ServerStrategy::InternalTcpPort => ServerBindConfig::TcpPort(self.internal_ip.clone()), // TODO: change to pod's internal port
+            ServerStrategy::ExternalTcpPort(_) => panic!("Cannot bind to external port"),
+            ServerStrategy::Demux(demux) => {
+                let mut config_map = HashMap::new();
+                for (key, underlying) in demux {
+                    config_map.insert(*key, self.server_config(underlying));
+                }
+
+                ServerBindConfig::Demux(config_map)
+            }
+            ServerStrategy::Merge(merge) => {
+                let mut configs = vec![];
+                for underlying in merge {
+                    configs.push(self.server_config(underlying));
+                }
+
+                ServerBindConfig::Merge(configs)
+            }
+            ServerStrategy::Tagged(underlying, id) => {
+                ServerBindConfig::Tagged(Box::new(self.server_config(underlying)), *id)
+            }
+            ServerStrategy::Null => ServerBindConfig::Null,
+        }
+    }
+
+    async fn copy_binary(&self, binary: &BuildOutput) -> Result<()> {
+        // Create a new pod in the running kubernetes cluster (we assume the user already has one up)
+        ProgressTracker::println("Copying binary to pod");
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        let file_name = format!("hydro-{}-binary", binary.unique_id);
+        // let metadata = std::fs::metadata("/Users/nickjiang/Nick/Hydro/hydroflow/hydro_deploy/core/src/kubernetes/hello_world_aarch64_musl")?;  // importantly, this file has executable permissions
+        {
+            let binary_data = binary.bin_data.clone();
+            let mut header = tar::Header::new_gnu();
+            header.set_path(file_name).unwrap();
+            header.set_size(binary_data.len() as u64);
+            // header.set_metadata(&metadata);
+            header.set_mode(0o755); // give the binary executable permissions
```

Comment:
I wonder if we could try doing this via `chmod` after `tar` is done, if that would help with the I/O errors.